app/logic/emailHandler2.py

```diff
@@ -0,0 +1,189 @@
+        # Email Template Data
+        self.template_identifier = self.raw_form_data['templateIdentifier']
+
+        if 'subject' in self.raw_form_data:
+            self.subject = self.raw_form_data['subject']
+
+        if 'body' in self.raw_form_data:
+            self.body = self.raw_form_data['body']
+
+        if 'replyTo' in self.raw_form_data:
+            self.reply_to = self.raw_form_data['replyTo']
+
+        # Event
+        if 'eventID' in self.raw_form_data:
+            event = Event.get_by_id(self.raw_form_data['eventID'])
+            self.event = event
+
+        # Program
+        if 'programID' in self.raw_form_data:
+            self.program_ids = self.fetch_event_programs(self.raw_form_data['programID'])
+
+        # Recipients
+        if 'recipientsCategory' in self.raw_form_data:
+            self.recipients_category = self.raw_form_data['recipientsCategory']
+            self.recipients = self.retrieve_recipients(self.recipients_category)
+
+        # Service Learning Course
+        if 'slCourseId' in self.raw_form_data:
+            self.sl_course_id = self.raw_form_data['slCourseId']
+
+    def fetch_event_programs(self, programId):
+        """ Fetches all the programs of a particular event """
+        # Non-student-led programs have "Unknown" as their id
+        # ---Q: maybe this id should be changed to something more specific?
+        if programId == 'Unknown':
+            programs = ProgramEvent.select(ProgramEvent.program).where(ProgramEvent.event==self.event.id)
+            return [program.program for program in programs.objects()]
+        else:
+            program = ProgramEvent.get_by_id(programId)
+            return [program.program]
+
+
+    def update_sender_config(self):
+        # We might need this.
+        # This functionality should be moved somewhere else.
+        # The function in another file would receive email_info[sender]
+        # and update the config based on that and wherever we will end up saving emails and passwords
+        #The sender information should be saved like so: {"name": [email, password], } or in the database
+        pass
+
+    def retrieve_recipients(self, recipients_category):
+        """ Retrieves recipient based on which category is chosen in the 'To' section of the email modal """
+        # Other potentioal recipients:
+        # - course instructors
+        # - course Participants
+        # - outside participants'
+        if recipients_category == "Interested":
+            recipients = (User.select()
+                .join(Interest)
+                .join(Program, on=(Program.id==Interest.program))
+                .where(Program.id.in_([p.id for p in self.program_ids])))
+
+        if recipients_category == "RSVP'd":
+            recipients = (User.select()
+                .join(EventParticipant)
+                .where(EventParticipant.event==self.event.id))

```

Comment:
If this category is RSVP, shouldn't it use EventRsvp instead of EventParticipant?

Reply:
Yes, it should be EventRsvp. We will change it.